File: 0413_arithmetic_slices/number_of_arithmetic_slices.java
class Solution {
    // Time: O(n)
    // Space: O(1)
    public int numberOfArithmeticSlices(int[] A) {
        int res = 0, curr = 0;
        for (int i = 2; i < A.length; i++) {
            if (A[i] - A[i - 1] == A[i - 1] - A[i - 2]) {
                curr += 1;
                res += curr;
            } else {
                curr = 0;
            }
        }
        return res;
    }
}

File: 0413_arithmetic_slices/number_of_arithmetic_slices.py
from typing import List


def numberOfArithmeticSlices(A: List[int]) -> int:
    """
    Time: O(n)
    Space: O(1)
    """
    res = curr = 0
    for i in range(2, len(A)):
        if A[i] - A[i - 1] == A[i - 1] - A[i - 2]:
            curr += 1
            res += curr
        else:
            curr = 0
    return res